vllm/config/parallel.py

```diff
@@ -223,6 +223,66 @@ class is dynamically inherited by the worker class. This is used to inject
         should only be set by API server scale-out.
     """
 
+    @field_validator("data_parallel_backend")
+    @classmethod
+    def _validate_data_parallel_backend(cls, data_parallel_backend: str) -> str:
+        if data_parallel_backend not in {"mp", "ray"}:
+            raise ValueError(
+                "data_parallel_backend must be either 'mp' or 'ray', "
+                f"but got {data_parallel_backend}."
+            )
+        return data_parallel_backend
```

Comment:
Can we use a `Literal` to type hint `data_parallel_backend` as in the other PRs so we don't need to manually validate the values?